core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -126,18 +84,37 @@ class BaseSnapshot implements Snapshot {
       Integer schemaId,
       List<ManifestFile> dataManifests) {
     this(
-        io,
         INITIAL_SEQUENCE_NUMBER,
         snapshotId,
         parentId,
         timestampMillis,
         operation,
         summary,
         schemaId,
-        null);
+        dataManifests.stream().map(ManifestFile::path).collect(Collectors.toList()));
```

Comment:
Since this is `Serializable`, we should use an array for the paths, rather than a list.